report/report.py

```diff
@@ -0,0 +1,213 @@
+from sisyphus import *
+import sys
+import os
+import getpass
+import pprint
+import gzip
+from typing import Dict, Union, Callable
+import time
+import json
+
+_Report_Type = Dict[str, Union[tk.Path, any]]
```

Comment:
Yeah this is where things (might) get ugly. In the end the way the job was structured from where I copied it it really depends on what your `report_format` function is doing. The usual input would maybe be something like `Union[tk.Variable, tk.Path, str]` or similar, but you could think of arbitrary things here. 
Actually this is where also other information like the keys come in, if the report makes use of them they are required.
The old jobs had a bunch of different formats and even more than one ReportJob (thats why it was named simple beforehand). The more I think about it maybe I will just remove a lot of stuff I deem irrelevant and then make more restrictions on the format of the report_format.

>Then if Variable is used, I don't see any more where get() is called.

This cost me like 2 hours and its hidden in the call of `str()` and `repr()` that are done within the reports (or by pprint).
https://github.com/rwth-i6/sisyphus/issues/104 this might even break things. But this just as a sidenote.

Reply:
But I think this is something that the report_format should handle, since this might be really user specific what the input is. So we could completely split it and say the `ReportJob` only gets a string it mails (then its just `MailJob`?) but I like that global information like the `config_path` and `sis_command_line` are actually set by the job since otherwise thats something the user would always have to deal with. And I am not sure if passing a string would require a job to produce this list since otherwise I think we would just end up with a lot of "UnifinishedVariable" in the report.